Build/classe cliente

Mat780 opened the pull request:
Está pull request é para mergear a branch Build/classe cliente com a main

A branch em questão é a respeito da criação das classes: Cliente, Data, ValidadorCPF e suas excessões
além de um Enum: UnidadeFederativa, para compor cliente.

Mat780 committed:
build: Adiciona gitignore para ignorar a pasta bin

Mat780 committed:
build: Adiciona excessões para as classes Cliente e Data

Mat780 committed:
build: Cria validador de CPF

Mat780 committed:
build: Atualiza classe Cliente e Data

GuilhermeAlmeida10000 commented on src/classes/Cliente.java:
```diff
@@ -0,0 +1,215 @@
+	
+	private String CPF;
+	private String nome;
+	private String rua;
```
K

GuilhermeAlmeida10000 commented on src/classes/Cliente.java:
```diff
@@ -0,0 +1,215 @@
+	private int numeroResidencia;
+	
+	private UnidadeFederativa UF;
+	
```
Sobre a Classe Cliente:
Linha 19: Aqui poderia ter um atributo separado para número da rua, pra ter uma diferenciação do endereço.
Linha 21: Não sei se o campo cidade seria importante já que os clientes em sua maioria residem em campo grande.
Faltam os campos (baseado no mockup):
- RG

Mat780 replied:
Sobre a rua, não é o que está no slide, to seguindo o slide a risca, justamente pra não dar problema
Sobre o RG, não se aplica também pelo mesmo motivo, o que tem RG é o projeto principal esse é só o caso de uso CadastrarCliente dos slides

GuilhermeAlmeida10000 reviewed (commented):
Coloquei um comentário na classe cliente, mas de resto, tá show de pelota.

GuilhermeAlmeida10000 reviewed (approved):
Achei massa demais essa classe de validação. Boa Matheus !

GuilhermeAlmeida10000 commented on src/excessoes/ExcessaoSomenteLetrasPermitidas.java:
```diff
@@ -0,0 +1,7 @@
+
+public class ExcessaoSomenteLetrasPermitidas extends RuntimeException{
+	public ExcessaoSomenteLetrasPermitidas() {
+		super("Neste campo não é permitido caracteres especiais ou números, somente letras");
```
Não são permitidos ...

GuilhermeAlmeida10000 reviewed (changes requested):
Nas exceções de Cliente, as mensagens podem ser mais curtas, algo como "Idade Inválida" ou "Cadastro não permitido". Na exceção telefone, dá pra simplificar a mensagem como "Formato de Telefone Inválido"

Mat780 commented:
> Nas exceções de Cliente, as mensagens podem ser mais curtas, algo como "Idade Inválida" ou "Cadastro não permitido". Na exceção telefone, dá pra simplificar a mensagem como "Formato de Telefone Inválido"

Ok alterarei isso hoje a noite

Mat780 committed:
style: Altera nomes de excessao para excecao

Mat780 commented:
@GuilhermeAlmeida10000 Alterado os nomes de excessao para excecao no ultimo commit

Mat780 committed:
style: Adequamento de nomes de variaveis

Mat780 committed:
refactor: Unifica excecao da classe Data

Haviam duas exceções muito similirares, elas foram juntadas e deram
origem a classe ExcessaoAnoInvalido

GiovannaRMendes committed:
style: Corrige os erros gramaticais

GiovannaRMendes reviewed (approved):
LGTM

GuilhermeAlmeida10000 reviewed (approved):
Lindo !
